metrics_sdk/lib/opentelemetry/sdk/metrics/state/metric_stream.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module State
+        class MetricStream
+          attr_reader :name, :description, :unit, :instrument_kind, :resource, :instrumentation_library, :data_points
+
+          def initialize(
+            name,
+            description,
+            unit,
+            instrument_kind,
+            resource,
+            instrumentation_library
+          )
+            @name = name
+            @description = description
+            @unit = unit
+            @instrument_kind = instrument_kind
+            @resource = resource
+            @instrumentation_library = instrumentation_library
+
+            @data_points = {}
+            @mutex = Mutex.new
+          end
+
+          def update(measurement)
+            @mutex.synchronize do
+              if @data_points[measurement.attributes]
+                @data_points[measurement.attributes] = @data_points[measurement.attributes] + measurement.value
+              else
+                @data_points[measurement.attributes] = measurement.value
+              end
```

Comment:
Is this supposed to be an array of metrics updates? Or just a running sum? I think we've implemented the latter here, but I was expecting the former based on the class name.

(Of course maybe you haven't gotten around to it yet. Or maybe I just misunderstand what we're doing entirely, which wouldn't surprise me at all. 😆 )